addons/account/models/company.py

```diff
@@ -509,8 +510,15 @@ def build_move_info(move):
             previous_hash = u''
             start_move_info = []
             hash_corrupted = False
+            current_hash_version = 1
             for move in moves:
-                if move.inalterable_hash != move._compute_hash(previous_hash=previous_hash):
+                move.invalidate_cache(['string_to_hash'])
```

Comment:
Just `moves.invalidate_cache(['string_to_hash'])` before the loop. If you don't specify the `ids`, it will invalidate the cache for all records, which is what you want

Reply:
Actually I don't think we want this because of https://github.com/odoo/odoo/pull/112320#discussion_r1107130173

Reply:
Right, but then you need to only disable the cache for this record otherwise I think you will recompute the value for each move at each iteration.
It is maybe easier to refactor the method to not use the cache at all?

Reply:
Yup it's done only for the current `move` (and only for the `string_to_hash` field), not `moves` if that's what you mean.

I actually thought of transforming `_compute_string_to_hash` into a getter instead of a compute since `string_to_hash` is only used in `_compute_hash`. But wouldn't the deletion of a method go against our stable policy (we should avoid modifying the signature, let alone deleting a full method) ?

Reply:
> Yup it's done only for the current `move` (and only for the `string_to_hash` field), not `moves` if that's what you mean.

No it isn't since you didn't provide the `ids` argument

> But wouldn't the deletion of a method go against our stable policy (we should avoid modifying the signature, let alone deleting a full method) ?

You can just extract the method and call it in the compute function

Reply:
Oh right !

Moving `_compute_string_to_hash`'s body inside `_compute_hash` ? But that would delete `_compute_string_to_hash` which is not OK.
Or doing transforming it into a getter and calling it in `_compute_hash` ? 
In either case, it would be similar to how it was here ? https://github.com/odoo/odoo/pull/112320#discussion_r1106861243

Reply:
> In either case, it would be similar to how it was here ? [#112320 (comment)](https://github.com/odoo/odoo/pull/112320#discussion_r1106861243)

Yes, but without the cache thing. What makes it weird is that we are using the cache of the ORM for something that can now change outside of the dependencies of the field.
If we wanted to do that, we could add a non stored field `hash_version` in the dependencies of `hash_string`

Reply:
Yes that was how it was done before (without the cache thingy). I'll redo that and add a better comment to explain why we must force the `compute_string_to_hash` inside `compute_hash`